genesyscloud/telephony_providers_edges_phone/genesyscloud_telephony_providers_edges_phone_proxy.go

```diff
@@ -152,20 +154,27 @@ func (p *phoneProxy) assignUserToStation(ctx context.Context, userId string, sta
 
 // getAllPhonesFn is an implementation function for retrieving all Genesys Cloud Phones
 func getAllPhonesFn(ctx context.Context, p *phoneProxy) (*[]platformclientv2.Phone, error) {
+	log.Printf("Entering the getAllPhonesFn method to retrieve all of the phone ids for export")
 	var allPhones []platformclientv2.Phone
+	const pageSize = 100
+	const sortBy = "id"
+
+	phones, response, err := p.edgesApi.GetTelephonyProvidersEdgesPhones(1, pageSize, sortBy, "", "", "", "", "", "", "", "", "", "", "", "", nil, nil)
+	if err != nil || (response != nil && response.StatusCode != http.StatusOK) {
+		log.Printf("getAllPhonesFn: error encountered while trying to get first page of phone data #%v statusCode: %d", err, response.StatusCode)
+		return nil, err
+	}
 
-	phones, _, _ := p.edgesApi.GetTelephonyProvidersEdgesPhones(1, 100, "", "", "", "", "", "", "", "", "", "", "", "", "", nil, nil)
+	log.Printf("Retrieved page 1 of %d pages of phone data.  Total number of phone records is %d", phones.PageCount, phones.Total)
 	for _, phone := range *phones.Entities {
```

Comment:
We might need make sure `Entities` is not nil. I have had problems before if no objects exist in the org `Entities` is equal to nil and I get a nil pointer exception.